### PR TITLE
#53 feature:Wishlist Management Functionalities

### LINKED ISSUE #27: #issue - After updating the product, we no longer see it vendor's collection

### PR DESCRIPTION
**Pull request by faid-terence**
#### What does this PR do?
Adds Buyer Wishlist Management Functionalities

#### Description of Task to be completed?

> Acceptance criteria:

GIVEN that the buyer is on a product page

WHEN the buyer clicks the ""Add to Wishlist"" button or a button to add the item to the wishlist without leaving the current page (optional)
THEN the item should be added to the buyer's wishlist and the buyer should see a confirmation message

#### How should this be manually tested?
```
git clone https://github.com/atlp-rwanda/ecommerce-app-predators-fe.git
cd ecommerce-app-predators-fe
git checkout branch-ft-buyer-manages-wishlist-53
npm i
npm run dev
```

#### What are the relevant pivotal tracker/Trello stories?
Trello [Link](https://trello.com/c/gBoJ48jP/53-a-buyer-should-be-able-to-manage-their-wishlist) (#-53)

#### Screenshots (if appropriate)
wishlist

### CODE REVIEW

**Review by yobuir — APPROVED** [review]

**Review by nili-alu — APPROVED** [review]

File: src/components/sample.tsx [hunk 1]
```diff
@@ -1,3 +1,3 @@
- export const add = (a: number, b: number) => {
+export const add = (a: number, b: number) => {
```
Comment (chanceDira) [inline src/components/sample.tsx:1]: I think we are not using this file, we can remove it.

File: src/components/table/UserTable.tsx [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
-import { useDispatch, useSelector } from "react-redux";
-import { disableAccount, getAllUsers } from "../../redux/action/disableAction";
-import { ToastContainer, toast } from "react-toastify";
+import { useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { disableAccount, getAllUsers } from '../../redux/action/disableAction';
+import { ToastContainer, toast } from 'react-toastify';
 import 'react-toastify/dist/ReactToastify.css';
 interface User {
   getAllUsers: any;
```
Comment (chanceDira) [inline src/components/table/UserTable.tsx:7]: use interface instead of any

File: src/components/wishlist/WishlistTable.tsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  deleteFromWishlist,
+  getWishlist,
+} from '../../redux/action/WishlistAction';
+import { useEffect } from 'react';
+import { ToastContainer } from 'react-toastify';
+interface Wishlist {
+  [x: string]: number;
+  Product: any;
```
Comment (chanceDira) [inline src/components/wishlist/WishlistTable.tsx:10]: I guess this should be a string instead of any

File: src/redux/action/productActions.tsx [hunk 4]
```diff
@@ -1,23 +1,35 @@
-import axios from "axios";
-import { createAsyncThunk } from "@reduxjs/toolkit";
+import axios from 'axios';
+import { createAsyncThunk } from '@reduxjs/toolkit';
 //import { RootState } from "../store/productStore";
 
+<<<<<<< HEAD
```
Comment (chanceDira) [inline src/redux/action/productActions.tsx]: fix this rebase issue

File: src/redux/store/Store.tsx [hunk 5]
```diff
@@ -15,10 +15,19 @@ import profileReducer from '../reducers/ProfileReducer';
 import productCollectionReducer from '../reducers/productReducer';
 import SetRolesSlice from '../reducers/SetRolesSlice';
 import searchReducer from '../reducers/Searchslice';
+<<<<<<< HEAD
```
Comment (chanceDira) [inline src/redux/store/Store.tsx]: rebasing issue

**Review by chanceDira — CHANGES_REQUESTED** [review]
Below are some requested changes, Also try to fix CI, they are failing.

File: src/components/table/UserTable.tsx (hunk 2 above)
Comment (faid-terence) [reply src/components/table/UserTable.tsx:7]: > use interface instead of any

I think this is Alvin's work ! not mine

**Review by vanessa-abewe — APPROVED** [review]

File: .github/workflows/slack.yml [hunk 6]
```diff
@@ -5,17 +5,20 @@ name: slack
 
 on:
   pull_request:
+<<<<<<< HEAD
```
Comment (chanceDira) [inline .github/workflows/slack.yml]: rebase issue

**Review by chanceDira — CHANGES_REQUESTED** [review]
Take a look on the below issue

**Review by chanceDira — APPROVED** [review]
Looking good

**Pull request merged**